src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -88,6 +114,20 @@ private static string GetDefaultNamespace()
 
         public HashSet<MethodInfo> ExistingModelFactoryMethods { get; }
 
+        private SignatureType? _signatureType;
+        public override SignatureType? SignatureType
+        {
+            get
+            {
+                // This can only be used for Mgmt now, because there are custom/hand-written code in HLC can't be loaded into CsharpType such as generic methods
+                if (!Configuration.AzureArm)
```

Comment:
since we potentially need to support DPG, we may need a TODO item here, and the check could be
`if (!Configuration.Generate1ConvenientClient)` which means "not HLC"